pyproject.toml

```diff
@@ -32,7 +32,11 @@ python = "^3.7"
 # Mandatory dependencies
 methoddispatch = "^3.0.2"
 msgpack = "^1.0.0"
-httpx = { version = "^0.24.1", extras = ["http2"] }
+httpx = [
+  { version = "^0.24.1", python = "~3.7" },
+  { version = "^0.25.0", python = "^3.8" },
+]
+h2 = "^4.1.0" # required for httx package, HTTP2 communication
```

Comment:
Can we not just install this through httpx as an extra?

Reply:
Seems we can't. It somehow fails : (

Reply:
Can you try the same from your end

Reply:
What does it fail with? The problem I have with this approach is that we'll have to manually keep the `h2` dependency in sync with new versions of httpx :/

Reply:
Did you try:
```
{ version = "^0.25.0", python = "^3.8" , extras = ["http2"]}
```

Reply:
I did. It doesn't work. It somehow fails to identify right `httpx` version for python 3.7

Reply:
OK I'll take a look

Reply:
I have checked the dependency `h2` internally, and it's always `h2 = >=3 && <5`

Reply:
Okay, I have added it as a part of new commit. It was failing me locally, I can check if it fails the same on CI

Reply:
Yeah it's failing for me locally too. Weird error message:
```
Because no versions of httpx match >0.25.0,<0.25.1 || >0.25.1,<0.25.2 || >0.25.2,<0.26.0
 and httpx (0.25.2) depends on httpx (0.25.2), httpx (>0.25.0,<0.25.1 || >0.25.1,<0.26.0) requires httpx (0.25.2).
And because httpx (0.25.1) depends on httpx (0.25.1)
 and httpx (0.25.0) depends on httpx (0.25.0), httpx (>=0.25.0,<0.26.0) requires httpx (0.25.0 || 0.25.1 || 0.25.2).
So, because ably depends on both httpx (^0.24.1) and httpx (^0.25.0), version solving failed.
```

Reply:
I checked and both of `0.24.x` and `0.25.x` are using same version of `h2` and that is `4.1.0`. Poetry shouldn't show this error

Reply:
It took lot of my time figuring out there might be something wrong with poetry itself : (

Reply:
Okay 👍

Reply:
Done!